### PR TITLE
SUS-2420 | import discussion-alerts.php script

### PR DESCRIPTION
**Pull request by macbre**
Import it from https://github.com/Wikia/community-engineering-misc/blob/master/discussion-alerts/discussion-alerts.php

https://wikia-inc.atlassian.net/browse/SUS-2420

### CODE REVIEW

File: maintenance/wikia/discussion-alerts.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+					list( $namespaceText, $parentPageTitle ) = explode( ':', $parts[0] . '/@' . $parts[1] );
+					$title = GlobalTitle::newFromText( $parentPageTitle, $pageData['ns'], $pageData['wid'] );
+					$parentId = $title->getArticleID();
+					$pageUrl = "{$wikiUrl}{$namespaceText}:{$parentPageTitle}";
+					$pageReport = "{$wikiUrl}Thread:$parentId (Thread title: {$pageData['titleStrict']}; Comment triggered: {$wikiUrl}Thread:{$pageData['pageid']})";
+				} else {
+					$pageReport = "http://{$pageData['host']}/wiki/Thread:{$pageData['pageid']} (Thread title: {$pageData['titleStrict']})";
+				}
+
+				if ( isset( $threadReported[$pageUrl] ) ) {
```
Comment (WikiaTech) [inline maintenance/wikia/discussion-alerts.php:83]: MAJOR Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

File: maintenance/wikia/discussion-alerts.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+		$pages = $result['response']['docs'];
+
+		foreach ( $pages as $pageData ) {
+			$pageUrl = $pageReport = $pageData['url'];
+
+			if ( in_array( $pageData['ns'], $threadNamespaces ) ) {
+				$wikiUrl = "http://{$pageData['host']}/wiki/";
+				$strippedPageTitle = str_replace( $wikiUrl, '', $pageData['url'] );
+				if ( substr_count( $strippedPageTitle, '@comment' ) === 2 ) {
```
Comment (WikiaTech) [inline maintenance/wikia/discussion-alerts.php:72]: MAJOR Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

File: maintenance/wikia/discussion-alerts.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+	while ( $i + $perpage < $numFound || $i == 1 ) {
+		echo "Fetching result...\n";
+		$result_raw = file_get_contents( $finalUrl . $i );
+
+		if ( empty( $result_raw ) ) {
+			die( "Something went horribly wrong. No response from Solr.\n" );
+		} else {
+			echo "Processing result...\n";
+			eval("\$result = " . $result_raw . ";" );
```
Comment (WikiaTech) [inline maintenance/wikia/discussion-alerts.php:59]: CRITICAL Remove this use of the "eval" function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1523)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by Grunny — APPROVED** [review]

**Pull request merged**